core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/utils/UrlHelper.kt

```diff
@@ -0,0 +1,25 @@
+package com.pierfrancescosoffritti.androidyoutubeplayer.core.player.utils
+
+/**
+ * Extracts the video ID from a YouTube URL.
+ *
+ * @param url The YouTube URL.
+ * @return The video ID, or the original URL if the video ID cannot be found.
+ */
+fun getVideoIdFromUrl(url: String): String? {
```

Comment:
This method should probably use java.net.URL class.

Reply:
we normally will get data from api or any source as string url, so I think we should go with string rather than make our lib user to convert it to URL instance

Reply:
The method should use URL internally to parse the string. Manual parse is error prone.

Reply:
could you share with me possible errors can be occur?
I think we do not need it.